clang/test/CIR/CodeGen/HIP/address-spaces.cpp

```diff
@@ -0,0 +1,19 @@
+#include "../Inputs/cuda.h"
+
+// RUN: %clang_cc1 -triple=amdgcn-amd-amdhsa -x hip -fclangir \
+// RUN:            -fcuda-is-device -fhip-new-launch-api \
+// RUN:              -emit-cir %s -o %t.ll
+// RUN: FileCheck --check-prefix=CIR --input-file=%t.ll %s
+
+__global__ void fn() {
+  int i = 0;
+  __shared__ int j;
+  j = i;
+}
+
+// CIR: cir.global "private" internal dso_local addrspace(offload_local) @_ZZ2fnvE1j : !s32i
```

Comment:
Please also add `LLVM` checks for lowering here, and add `OGCG` for tracking classic codegen alongside (see `clang/test/CIR/CodeGen/CUDA/cuda-builtin-vars.cu` as an example). Same for the other tests!